src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -868,6 +901,19 @@ internal bool TryGetCollectionIterationType(ExpressionSyntax syntax, TypeSymbol
                 var elementType = elementTypeWithAnnotations.Type;
                 Debug.Assert(elementType is { });
 
+                if (collectionTypeKind == CollectionExpressionTypeKind.ImplementsIEnumerable)
+                {
+                    if (!HasCollectionExpressionApplicableConstructor(node.Syntax, targetType, diagnostics))
+                    {
+                        reportedErrors = true;
```

Comment:
>reportedErrors = true; [](http://example.com/codeflow?start=24&length=22)

Similar suggestion about reporting dedicated error instead.

Reply:
On the other hand if "error CS1729: '{targetType}' does not contain a constructor that takes 0 arguments" and the like are the only errors that we can get from ```HasCollectionExpressionApplicableConstructor```, that is probably fine too.